pkg/fqdn/lookup/endpoint.go

```diff
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package lookup
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/netip"
+
+	"github.com/cilium/cilium/pkg/endpoint"
+	"github.com/cilium/cilium/pkg/endpointmanager"
+	"github.com/cilium/cilium/pkg/identity"
+	"github.com/cilium/cilium/pkg/ipcache"
+	"github.com/cilium/cilium/pkg/node"
+)
+
+type ProxyLookupHandler interface {
+	// LookupSecIDByIP is a provided callback that returns the IP's security ID
+	// from the ipcache.
+	LookupSecIDByIP(ip netip.Addr) (secID ipcache.Identity, exists bool)
+
+	// LookupByIdentity is a provided callback that returns the IPs of a given security ID.
+	LookupByIdentity(nid identity.NumericIdentity) []string
+
+	// LookupRegisteredEndpoint looks up the endpoint corresponding
+	// to a given IP address. It correctly handles *all* IPs belonging to the node, not just that
+	// of the node endpoint.
+	LookupRegisteredEndpoint(endpointAddr netip.Addr) (endpoint *endpoint.Endpoint, isHost bool, err error)
+}
+
+type proxyLookupHandler struct {
+	ipCache         *ipcache.IPCache
+	localNodeStore  *node.LocalNodeStore
+	endpointManager endpointmanager.EndpointManager
+}
+
+var _ ProxyLookupHandler = &proxyLookupHandler{}
+
+func (p *proxyLookupHandler) LookupRegisteredEndpoint(endpointAddr netip.Addr) (endpoint *endpoint.Endpoint, isHost bool, err error) {
+	if e := p.endpointManager.LookupIP(endpointAddr); e != nil {
+		return e, e.IsHost(), nil
+	}
+
+	localNode, err := p.localNodeStore.Get(context.TODO())
```

Comment:
nit: `context.Background()` if you're not interested in cancellation? I think this will probably change with some refactoring of the local node store soon anyway.